### ISSUE TITLE
DateTypeHandler is slow

### ISSUE THREAD

**Issue by GoogleCodeExporter** [unlockha/beanio#99]
```
What steps will reproduce the problem?  If applicable, please provide a
mapping configuration and sample record input to recreate the problem.

I am parsing dates actively and DateTypeHandler takes a lot of CPU according to 
the profiler. It looks like it creates new DateFormat for each parse and it 
takes a lot of time.  

I've replaced it with simple format that reuses parser and it helps a lot.

What version of BeanIO are you using? What JDK version?

BeanIO 2.1.0.M2, jdk1.7.0_45

```

Original issue reported on code.google.com by `tiv...@gmail.com` on 9 Jan 2014 at 3:16

**Comment by GoogleCodeExporter** [comment]
```
Agreed.  I fixed this for number types in 2.0.1.M2 and plan to do the same for 
date types in the next release.
```

Original comment by `kevin.s...@gmail.com` on 11 Jan 2014 at 8:11
- Changed state: **Accepted**

**Comment by GoogleCodeExporter** [comment]
```
Fixed for release 2.1.0.  Snapshot JAR attached.
```

Original comment by `kevin.s...@gmail.com` on 22 Jan 2014 at 3:46
- Added labels: **Milestone-2.1.0**

Attachments:
- [beanio-2.1.0.SNAPSHOT.jar](https://storage.googleapis.com/google-code-attachments/beanio/issue-99/comment-2/beanio-2.1.0.SNAPSHOT.jar)

**Comment by GoogleCodeExporter** [comment]
Original comment by `kevin.s...@gmail.com` on 7 Sep 2014 at 2:16
- Changed state: **Fixed**
